src/libbson/src/bson/bson-decimal128.c

```diff
@@ -581,10 +581,12 @@ bson_decimal128_from_string_w_len (const char *string,     /* IN */
 #else
 #define SSCANF sscanf
 #endif
-      int read_exponent = SSCANF (++str_read, "%d%n", &exponent, &nread);
```

Comment:
What was the cause of the issue? Was `SSCANF` not returning an error when parsing integer literals outside of the range of INT32?

That may help to determine if a problem would occur with an exponent outside of the range of INT64.

Reply:
SSCANF was parsing integers without error, but would overflow as such:
```
+2147483648 -> -2147483648 // min 32 bit integer
+2147483649 -> -2147483647
+2147483650 -> -2147483646

-2147483649 -> 2147483647 // max 32 bit integer
-2147483650 -> 2147483646
-2147483651 -> 2147483645
```
This behavior is demonstrated in this branch https://github.com/mongodb/mongo-c-driver/compare/master...joshbsiegel:mongo-c-driver:cdriver-4662-demo?expand=1 by running `./cmake-build/src/libmongoc/test-libmongoc -d --no-fork -l "/bson/test_overflowing_exponent"`